### ISSUE TITLE
Asset Manager for web versions of TheXTech

### ISSUE THREAD

**Issue by Superbloxen** [Wohlstand/TheXTech#694]
One issue with the web version of the game was that levels made in the editor would be deleted when the tab gets closed. In order to work around this, an export feature could be added to allow for the saving of levels to a file, and an import feature to edit an existing level. This could also work for save files to prevent frustration when a browser's data gets cleared.  Additionally, an asset manager could also be used to run episodes externally (from a file), and save data directly to a machine.

**Comment by ds-sloth** [comment]
Great ideas. All of these will rely on similar mechanisms, but we may want to implement them separately:
- [ ] Way to export user-created content from the editor
- [ ] Way to import episodes
- [ ] Way to export / import save data and settings

> deleted when the tab gets closed

I agree that this definitely shouldn't happen. Have you set up your browser to delete cookies / cache frequently? If so this is unavoidable, but if not, we should open a separate issue about that.

**Comment by Superbloxen** [comment]
>  Have you set up your browser to delete cookies / cache frequently? 

No. Save data is unaffected by closing the browser, but only custom levels/episodes are erased.

**Comment by ds-sloth** [comment]
Wow, that's really not good. I'll try to fix this in the next few days, then we can develop the other features you requested more slowly.

**Comment by ds-sloth** [comment]
Commit 616ab10d should have fixed the bug where custom levels/episodes got lost. Can you please confirm this on your side?

(Note: the web app update process is a bit complicated currently, and I'm planning to improve it soon. Right now, you need to open the page, wait long enough for it to update in the background, _close the page_ -- don't just refresh it -- and reopen it.)

**Comment by Superbloxen** [comment]
Apologies for the exceedingly late reply. The level/episode deletion issue is fixed now.

**Comment by ds-sloth** [comment]
Excellent! I'll leave this issue open so we can track the useful new features you requested.

**Comment by Wohlstand** [comment]
As I understand, we should take some library that can pack/unpack ZIP archives on the fly, and use it to generate the ZIP archive from idbfs content and let browser save it (I have some sketched written for some work needs), and import files into idbfs as both plain or ZIP-packed form.

I think, this needs a kind of "file manager" for the idbfs stuff, so, user can manipulate files and adjust some stuff at the idbfs directly. That's should be implemented pretty simple.

**Comment by Wohlstand** [comment]
First that I found, this: https://github.com/bpajk/WasmZip
